### PR TITLE
Publication Landscape and Document parser update

### PR DESCRIPTION
**Pull request by danich1**
This PR is generated in response to the conversation in #81. The document parser was not converting words into their base form (lemmatization e.g. walked -> walk) and it was suggested that we incorporate that for generating document vectors. As a result the publication landscape for the server changed along with the SAUCIE model and publication paper dataset (uploaded to google cloud in a folder called updated-data). The KNN model will need to be re-sharded again as a heads up @dongbohu.

### CODE REVIEW

**Comment by vincerubinetti** [comment]
It looks like the bin width has changed here.

image

I forget how we did this, but there's some variable in the javascript code you have to match to whatever the bin width was when you exported the plot.

**Comment by vincerubinetti** [comment]
Here it is: https://github.com/greenelab/preprint-similarity-search/blob/master/frontend/src/map.js#L13

**Comment by vincerubinetti** [comment]
Bin size of `0.65` doesn't look right. It looks like it's more like `0.4`:
image

Also you ran the plot through `filter-plot.html` right? I think you did otherwise it wouldn't show anything, but on Netlify the graph was taking a while to load and the filter was supposed to fix that.

Is this plot data going to be updated more frequently now? If so I will probably want to figure out a way to pre-process this automatically.

**Comment by danich1** [comment]
>Bin size of 0.65 doesn't look right. It looks like it's more like 0.4:

Ah. Right now I understand why we were using bin_width. I changed my code to use R's auto binwidth generator, but that's not right. Will default back for that plot. It will be 0.85 binwidth.

>Also you ran the plot through filter-plot.html right? I think you did otherwise it wouldn't show anything, but on Netlify the graph was taking a while to load and the filter was supposed to fix that.

Yeah I ran the new plot through your filter-plot html code and updated the plot.json as well as the main json plot file as well.

File: server/word_vectors.py [hunk 1]
```diff
@@ -49,17 +49,22 @@ def parse_content(content, xml_file=True):
...
+        all_text = " ".join(all_text)
 
-        for text in text_to_process:
-            tokens = list(map(str, nlp(text[0])))
+        all_tokens = list(
+            map(
+                lambda x: str(x.lemma_),
+                filter(
+                    lambda tok: str(tok.lemma_) in model.wv
+                    and str(tok.lemma_) not in nlp.Defaults.stop_words,
```
Comment (dongbohu) [inline server/word_vectors.py:61]: Isn't `tok.lemma_` already a string?

File: server/word_vectors.py [hunk 2]
```diff
@@ -70,12 +75,11 @@ def parse_content(content, xml_file=True):
 
         # Convert text to word vectors and continue processing
         for page in pdf_parser:
-            tokens = list(map(str, nlp(page.getText())))
 
             word_vectors += [
-                word_model_wv[tok]
-                for tok in tokens
-                if tok in word_model_wv and tok not in nlp.Defaults.stop_words
+                word_model_wv[str(tok.lemma_)]
```
Comment (dongbohu) [inline server/word_vectors.py:80]: `str()` conversion seems redundant.

File: server/word_vectors.py [hunk 3]
```diff
@@ -70,12 +75,11 @@ def parse_content(content, xml_file=True):
...
+                for tok in nlp(page.getText())
+                if str(tok.lemma_) in word_model_wv and str(tok.lemma_) not in nlp.Defaults.stop_words
```
Comment (dongbohu) [inline server/word_vectors.py:82]: Redundant `str()` conversion.

**Review by dongbohu — COMMENTED** [review]
Left a few comments.

File: server/word_vectors.py (hunk 1 above)
Comment (danich1) [reply server/word_vectors.py:61]: Yes good call.

**Review by dongbohu — APPROVED** [review]

**Review by vincerubinetti — APPROVED** [review]

**Comment by dongbohu** [comment]
I am merging/closing this PR so that the frontend can use the latest plot json file.

**Pull request merged**